samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -89,6 +89,11 @@ public GroupByContainerCount(int containerCount) {
 
     validateTasks(tasks);
 
+    if (localityManager == null) {
```

Comment:
I just made a comment in the JIRA regarding using `job.host-affinity.enabled` config. If you use that config, that can be passed from the factory to the GroupByContainerCount class. If host-affinity is enabled and localityManager is not defined, then it should throw an exception.